### PR TITLE
fix(component/datalist): checking if value exist on update

### PR DESCRIPTION
**Pull request by jlame**
**What is the problem this PR is trying to solve?**
When typing search if the input was blurred or enter key pressed, the search value was updated as value whereas it does not exist in the suggestions.

**What is the chosen solution to this problem?**
If the value doesn't exist, it is updated with the previous selected value or by empty string.
**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/components.eslint.txt [hunk 1]
```diff
@@ -8,6 +8,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   1:1  error  '@storybook/react' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   4:2  error  Unexpected require()                                                                    global-require
 
+/home/travis/build/Talend/ui/packages/components/src/Datalist/Datalist.component.test.js
+  148:9  error  'payload' is assigned a value but never used  no-unused-vars
```
Comment (jsomsanith) [inline output/components.eslint.txt]: Please fix linter error

File: packages/components/src/Datalist/Datalist.component.js [hunk 2]
```diff
@@ -212,8 +211,22 @@ class Datalist extends Component {
 					}
 				}
 			}
-			const payload = { value: get(enumValue, 'value', value) };
-			this.props.onChange(event, payload);
+			const payload = { value: get(enumValue, 'value') };
+			if (payload.value || !this.props.restricted) {
+				this.props.onChange(
+					event,
+					payload.value ? payload : { value: get(enumValue, 'value', value) },
```
Comment (jsomsanith) [inline packages/components/src/Datalist/Datalist.component.js]: This block is quite hard to read and you fetch the enumValue twice. What about
```
const selectedEnumValue = get(enumValue, 'value');
if (selectedEnumValue || !this.props.restricted) {
	this.props.onChange(
		event,
		payload: { value:  selectedSuggestionValue || value }
	);
	this.setState({
		previousValue: typeof previousValue === 'object' ? previousValue.title : previousValue,
	});
}
```

File: packages/components/src/Datalist/Datalist.component.js [hunk 3]
```diff
@@ -212,8 +211,22 @@ class Datalist extends Component {
...
+				);
+				// setting the previous value to current only if the current value exists
+				this.setState({
+					previousValue: typeof previousValue === 'object' ? previousValue.title : previousValue,
+				});
+			} else {
+				this.setState({
```
Comment (jsomsanith) [inline packages/components/src/Datalist/Datalist.component.js]: You should call `resetValue()`

File: packages/components/src/Datalist/Datalist.component.test.js [hunk 4]
```diff
@@ -111,15 +111,69 @@ describe('Datalist component', () => {
...
+		// when
+		input.simulate('blur');
+
+		// then
+		const payload = { value: 'foobar' };
+		expect(onChange).toBeCalledWith(expect.anything(), payload);
+		expect(wrapper.find(Typeahead).props().items).toBe(null);
+	});
+
+	it('should reset suggestions and change value on blur when not in restricted mode', () => {
```
Comment (jsomsanith) [inline packages/components/src/Datalist/Datalist.component.test.js]: This one is the exact same as the previous one, you can remove it.

Comment (jlame) [reply packages/components/src/Datalist/Datalist.component.test.js]: not exactly, this one is to check that the value is selected even if it does not exist in the suggestions

**Pull request merged**